File: Selection_sort_JAVA/SelectionSort.java
//Made By NAMAN

package selectionsort;


public class SelectionSort {
    public static void SelectionSort(int a[]){
        for(int i=0;i<a.length;i++){
            int pos=i;
            for(int j=i+1;j<a.length;j++){
                if(a[pos]>a[j]){
                    pos=j;
                }
                if(pos!=i){
                    int temp=a[i];
                    a[i]=a[pos];
                    a[pos]=temp;
                }
            }
        }
    }

    public static void main(String[] args) {
        int a[]={77,32,87,12,99,1};
        System.out.println("Before Selection Sort");  
        for(int i=0;i<a.length;i++){  
            System.out.print(a[i]+" ");  
        }  
        System.out.println();  
          
        SelectionSort(a);//sorting array using selection sort  
         
        System.out.println("After Selection Sort");  
        for(int i:a){  
            System.out.print(i+" ");  
        }  
    }  
    
    
}

File: selection_sort/C/selection_sort.c
#include <stdio.h>	// for printf function

void selection_sort(int[], int);
void print_array(int[], int);

int main() {
	int arr[] = {4, 3, 42, 82, 5, 2, 33};
	int n = sizeof(arr) / sizeof(arr[0]);
	selection_sort(arr, n);
	printf("Sorted array is: ");
	print_array(arr, n);
	printf("\n");
	return 0;
}

/* Function to print array */
void print_array(int arr[], int n) {
	for (int i = 0; i < n; ++i) {
		printf("%d ", arr[i]);
	}
}

/* Selection Sort algorithm */
void selection_sort(int arr[], int n) {
	for (int i = 0; i < n - 1; ++i) {
		int small = i;
		for (int j = i + 1; j < n; ++j) {
			if (arr[j] < arr[small])
				small = j;
		}
		int temp = arr[i];
		arr[i] = arr[small];
		arr[small] = temp;
	}
}
